Libraries/test/TestServerlessApp/SQSEventExamples/InvalidSQSEvents.cs.error

```diff
@@ -0,0 +1,41 @@
+﻿using Amazon.Lambda.Annotations;
+using Amazon.Lambda.Annotations.APIGateway;
+using Amazon.Lambda.Annotations.SQS;
+using Amazon.Lambda.SQSEvents;
+using System;
+
+namespace TestServerlessApp.SQSEventExamples
+{
+    public class InvalidSQSEvents
+    {
+        [LambdaFunction]
+        [SQSEvent("test-queue", BatchSize = 0, MaximumBatchingWindowInSeconds = 302, MaximumConcurrency = 1)]
+        public void ProcessMessageWithInvalidSQSEventAttributes(SQSEvent evnt)
+        {
+            Console.WriteLine($"Event processed: {evnt}");
+        }
+
+        [LambdaFunction]
+        [SQSEvent("test-queue")]
+        public void ProcessMessageWithInvalidParameters(SQSEvent evnt, bool invalidParameter1, int invalidParameter2)
+        {
+            Console.WriteLine($"Event processed: {evnt}");
+        }
+
+        [LambdaFunction]
+        [SQSEvent("test-queue")]
+        public bool ProcessMessageWithInvalidReturnType(SQSEvent evnt)
+        {
+            Console.WriteLine($"Event processed: {evnt}");
+            return true;
+        }
+
+        [LambdaFunction]
+        [RestApi(LambdaHttpMethod.Get, "/")]
```

Comment:
Can we not have this sample. It doesn't make logical sense because the event is between SQS and API Gateway or so similar. We get users asking for a single function that can handle events from both SQS and API Gateway but there isn't really a good way to implement that. This example my give a false sense that it could work.

Reply:
All functions (including this one) that are written in the `TestServerlessApp/SQSEventExamples/InvalidSQSEvents.cs.error` are invalid. This file is sent as input to the source generator unit tests and we assert that compilation errors are thrown with the appropriate diagnostic message.

In case of multiple events added to a Lambda method, we would throw a compilation error with the [MultipleEventsNotSupported](https://github.com/aws/aws-lambda-dotnet/blob/0558191fdff4235b5dbd2b2ba80633edd94c10ab/Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/Diagnostics/DiagnosticDescriptors.cs#L27) diagnostic message.

Refer to the `VerifyInvalidSQSEvents_ThrowsCompilationErrors` unit test added in this PR.

Reply:
You are right the file name is even called `InvalidSQSEvents.